src/client/components/PullRequestSummary/PullRequestSummary.js

```diff
@@ -128,77 +134,116 @@ export const RepositoriesSection = ({ pullRequest, paths } : {
-          <Col md={1}>
-            <span style={{ color: approvedColor, fontSize: '16px', marginRight: '10px' }}>
-              <i className="fa fa-check" aria-hidden="true" />
-            </span>
-          </Col>
-          <Col md={11}>
-            <div style={{ fontSize: '13px' }}>
-              Alexey Zakharov, Scott Bilas, Aras Pranckevičius, Tim Cooper
-            </div>
-          </Col>
-        </Row>
-        <Row>
-          <Col md={1}>
-            <span
-              style={{ color: 'rgb(128, 154, 206)', fontSize: '16px', marginRight: '10px' }}
-            >
-              <i className="fa fa-circle-o-notch fa-spin fa-fw" />
-            </span>
-          </Col>
-          <Col md={11}>
-            <div style={{ fontSize: '13px' }}>
-              Ante Ilic, Alex Lian
-            </div>
-          </Col>
-        </Row>
-        <Row>
-          <Col md={1}>
-            <span style={{ color: 'grey', fontSize: '16px', marginRight: '10px' }}>
-              <i className="fa fa-question" aria-hidden="true" />
-            </span>
-          </Col>
-          <Col md={11}>
-            <div style={{ fontSize: '13px' }}>
-              {props.pullRequest.reviewers.map(r => r.user.fullName).join(', ')}
-            </div>
-          </Col>
-        </Row>
-        {props.toggleReviewers &&
-          <Row style={{ paddingTop: '20px', paddingLeft: '50px' }}>
-            <Reviewers reviewers={prReviewers} onAdded={props.onAddReviewer} />
-          </Row>
-        }
-      </Col>
-      <Col md={1}>
-        <div
-          onClick={props.onToggleReviewers}
-          style={{ color: '#dbdedf', float: 'right', fontSize: '20px', cursor: 'pointer' }}
-        >
-          <i className="fa fa-pencil" aria-hidden="true" />
-        </div>
-      </Col>
-    </Row>
-  </ListGroupItem>
+const ReviewerList = ({ reviewers, icon, tooltip }) => {
+  if (reviewers.length === 0) {
+    return null
+  }
+  const tooltipOverlay = <Tooltip id="tooltip">{tooltip}</Tooltip>
+  return (
+    <li>
+      <span>
+        <OverlayTrigger placement="left" overlay={tooltipOverlay}>
+          <i
+            style={{ marginRight: '12px', width: '12px', marginBottom: '6px' }}
+            className={`fa ${icon}`}
+          />
+        </OverlayTrigger>
+        {reviewers.map(r => r.user.fullName).join(', ')}
+      </span>
+    </li>
+  )
+}
+
+
+export const ReviewersSection = (props: PullRequestSummaryProps) => {
+  const { reviewers } = props.pullRequest
+
+  // Lodash groupBy is not really what we want here, all groups should be represented.
+  type GroupsType = { [key: PullRequestReviewerStatusType]: Array<PullRequestReviewerType> }
+  const reviewerGroups : GroupsType = {
+    not_reviewed: reviewers.filter(r => r.status === 'not_reviewed'),
+    approved: reviewers.filter(r => r.status === 'approved'),
+    rejected: reviewers.filter(r => r.status === 'rejected'),
+    under_review: reviewers.filter(r => r.status === 'under_review'),
+  }
```

Comment:
I understand what you mean.

> So if you pass the hole pullRequest in the component it will rerender each time pullRequest is changed (even if just title changed)

No matter what, all components render all the time something changes, because we are not implementing `shouldComponentUpdate` for any components (except Breadcrumb)... So this is true for the entire application.

Re-rendering is super fast for these simple components, so I don't think the complexity of having this in a selector in another file is worth it. It is rather view specific that we need this grouping, do we need this other places? 

: )